common/headtracker/types/head.go

```diff
@@ -0,0 +1,17 @@
+package types
+
+import (
+	txmgrtypes "github.com/smartcontractkit/chainlink/v2/common/txmgr/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+//go:generate mockery --quiet --name Head --output ./mocks/ --case=underscore
+type Head[H types.Head[BLOCK_HASH], BLOCK_HASH types.Hashable, CHAIN_ID txmgrtypes.ID] interface {
+	types.Head[BLOCK_HASH]
+	// Equals returns true if the two heads are equal
+	Equals(H) bool
+	// ChainID returns the chain ID that the head is for
+	ChainId() CHAIN_ID
+	// Returns true if the head has a chain Id
+	HasChainId() bool
```

Comment:
nit/ capitalize `ID` acronym:
```suggestion
	ChainID() CHAIN_ID
	// Returns true if the head has a chain Id
	HasChainID() bool
```

Reply:
Thanks for the note! I thinking hard on whether to go camel case or capitalise them 😂